protocol/contracts/core/OtcOffer.sol

```diff
@@ -0,0 +1,168 @@
+    uint256 public immutable scalar;
+
+    /// @notice Any update to the `offerPrice` must be within this percentage (as basis points. 20% = 2_000) 
+    /// of the existing `offerPrice`.
+    uint256 public offerPriceUpdateThresholdBps;
+
+    /// @notice 100% represented as basis points.
+    uint256 public constant BPS_100_PCT = 10_000;
+
+    event OfferPriceSet(uint256 _offerPrice);
+    event OfferPriceUpdateThresholdBpsSet(uint256 offerPriceUpdateThresholdBps);
+    event Swap(address indexed account, address indexed fundsOwner, uint256 userSellTokenAmount, uint256 userBuyTokenAmount);
+    event FundsOwnerSet(address indexed fundsOwner);
+
+    error OfferPriceDeltaTooBig(uint256 newOfferPrice, uint256 maxPriceAllowed);
+
+    constructor(
+        address _userSellToken,
+        address _userBuyToken,
+        address _fundsOwner,
+        uint256 _offerPrice,
+        OfferPricingToken _offerPricingToken,
+        uint256 _offerPriceUpdateThresholdBps
+    ) {
+        userSellToken = IERC20Metadata(_userSellToken);
+        userBuyToken = IERC20Metadata(_userBuyToken);
+        fundsOwner = _fundsOwner;
+
+        offerPrice = _offerPrice;
+        offerPricingToken = _offerPricingToken;
+        offerPriceUpdateThresholdBps = _offerPriceUpdateThresholdBps;
+
+        // The price is always specified in 18dp
+        // Eg If selling OHM (9dp) for USDC (6dp):
+        // 1000 OHM (9dp) * 11 USDC/OHM (18dp) = 11_000 USDC (6dp)
+        // So this would need to get scaled down by 30dp
+        uint256 scaleDecimals = offerPricingToken == OfferPricingToken.UserBuyToken
+            ? OFFER_PRICE_DECIMALS + userSellToken.decimals() - userBuyToken.decimals()
+            : OFFER_PRICE_DECIMALS + userBuyToken.decimals() - userSellToken.decimals();
+        scalar = 10 ** scaleDecimals;
+    }
+
+    /// @notice Owner can pause user swaps from occuring
+    function pause() external onlyOwner {
+        _pause();
+    }
+
+    /// @notice Owner can unpause so user swaps can occur
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+
+    /// @notice Owner can update where the funds are pulled from and sent to upon a swap
+    function setFundsOwner(address _fundsOwner) external onlyOwner {
+        if (_fundsOwner == address(0)) revert CommonEventsAndErrors.InvalidAddress();
+        fundsOwner = _fundsOwner;
+        emit FundsOwnerSet(_fundsOwner);
+    }
+
+    /// @notice Owner can update the offer price at which user swaps can occur
+    /// @dev The new price must be within the `offerPriceUpdateThresholdBps` threshold of
+    /// the existing price
+    function setOfferPrice(uint256 _offerPrice) external onlyOwner {
+        if (_offerPrice == 0) revert CommonEventsAndErrors.ExpectedNonZero();
+
+        uint256 existingOfferPrice = offerPrice;
+        uint256 absMaxDelta = existingOfferPrice * offerPriceUpdateThresholdBps / BPS_100_PCT;
+        if (_offerPrice > existingOfferPrice) {
```

Comment:
Why not:

```
   if (_offerPrice > existingOfferPrice + absMaxDelta || _offerPrice < existingOfferPrice - absMaxDelta) {
       revert OfferPriceDeltaTooBig(_offerPrice, maxValid);
   }
```
less variables, gas etc?

Reply:
Yeh - cheaper as 2 lines

Reply:
Also need to be sure it doesn't underflow if `absMaxDelta > existingOfferPrice` in the second branch. At which point I decided to split properly